modules/data-loading/pages/kafka-loader/kafka-ssl-sasl.adoc

```diff
@@ -0,0 +1,218 @@
+* Ensure you have link:https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_gssapi.html[configured SASL with GSSAPI] on the broker.
+* Ensure the IP address and the hostname of the broker is in the `/etc/hosts` file.
+
+.On client
+* Make sure the following dependencies are installed on the client (TigerGraph) server.
+** On Centos:
++
+[,console]
+----
+yum install krb5-workstation
+yum install cyrus-sasl-gssapi
+----
+** On Ubuntu/Debian
++
+[,console]
+----
+apt install krb5-user
+apt install libsasl2-modules-gssapi-mit
+apt install libsasl2-modules-gssapi-heimdal <1>
+----
+<1> You only need to install `libsasl2-modules-gssapi-heimdal` if you are on Ubuntu 20.04 LTS
+
+=== Procedure
+. Copy the following files from the server to the client's m1 machine.
+* The server's producer keytab.
+. Define the data source and provide SASL configurations in the data source configuration file.
+For example:
++
+[,json]
+----
+{
+   "broker": "kafka-0.tigergraph.com:9092",
+   "kafka_config":{
+       "security.protocol": "SASL_PLAINTEXT",
+       "sasl.mechanism": "GSSAPI",
+       "sasl.kerberos.service.name":"kafka",
+       "sasl.kerberos.principal": "kafka-producer@TIGERGRAPH.COM", <1>
+       "sasl.kerberos.keytab": "/home/tigergraph/kafka_ssl/kafka-producer.keytab" <2>,
+       "sasl.jaas.config": "com.sun.security.auth.module.Krb5LoginModule required  debug=true useKeyTab=true  storeKey=true  keyTab=\"/home/tigergraph/kafka_ssl/kafka-producer.keytab\"  principal=\"kafka-producer@TIGERGRAPH.COM\";"
+   }
+}
+----
+<1> `sasl.kerberos.principal` needs to match the principal value in the broker's JAAS configuration file.
+<2> `sasl.kerberos.keytab` needs to match the paths where you copied the producer's keytab.
+
+Once you have defined the data source with the correct SASL configuration, you have configured SASL with GSSAPI between TigerGraph and your Kafka cluster for Kafka loading.
+
+When you run loading jobs from this data source, the Kafka cluster will authenticate the identity of TigerGraph server.
+However, the data in transit remains unencrypted.
+
+== Set up SSL and SASL
+
+You can set up SASL authentication protocol over an SSL-encrypted communication channel.
+
+=== Before you begin
+* Follow Confluent documentation to configure link:https://docs.confluent.io/platform/current/kafka/authentication_sasl/authentication_sasl_gssapi.html#brokers[SASL with GSSAPI] on the broker, and specify `security.inter.broker.protocol=SASL_SSL` to be `SASL_SSL`.
+This guide focuses on configuring the client (TigerGraph server).
+
+=== Procedure
+. Copy the following files from the broker to the client.
+* The broker's CA
+* The client's certificate signed by the broker's CA
+* The client's key
+* The broker's truststore
+* The broker's keystore
+* The Kafka producer's keytab
+. On the client server, create a JAAS configuration file `kafka_client_jaas.conf`. In the configuration file, configure the following values:
```

Comment:
should add following line into `~/.bashrc` to specify the auth config file path.
export KAFKA_OPTS="-Djava.security.auth.login.config=/home/tigergraph/kafka_ssl/kafka_client_jaas.conf"